src/gm3/components/layers/vector.js

```diff
@@ -45,8 +45,18 @@ import getStyleFunction from 'mapbox-to-ol-style';
 function defineSource(mapSource) {
     if(mapSource.type === 'wfs') {
         // add a wfs type source
+        let format = GML2Format;
+        let output_format = 'text/xml; subtype=gml/2.1.2';
+
+        if(mapSource.params.outputFormat
+          && mapSource.params.outputFormat.indexOf('json') >= 0) {
+
+            format = GeoJSONFormat;
+            output_format = 'application/json';
```

Comment:
I agree that sniffing for 'json' in the outputFormat param is a good enough way to detect to use the GeoJSONFormat over GML, but why not `output_format = mapSource.params.outputFormat;`  Then people can make it work for servers expecting something else.  Technically, you need to give the server back what it offers in GetCapabilities (and as far as I can tell there isn't one single best answer for what the GeoJSON mime type should be yet.)